src/ai/backend/client/cli/app.py

```diff
@@ -163,16 +209,46 @@ def app(session_id, app, protocol, bind):
         nonlocal api_session, runner
         loop = current_loop()
         api_session = AsyncSession()
+        arguments, envs = {}, {}
+        if len(arg) > 0:
+            for argline in arg:
+                split = argline.strip().split(' ', maxsplit=2)
+                if len(split) == 2:
+                    arguments[split[0]] = split[1]
+                else:
+                    arguments[split[0]] = ''
```

Comment:
In this case, we need to be able to distinguish empty string argument and `None`.
How about apply `shlex.split` instead of manual parsing so that users may specify empty string arguments?

A full example:
```console
> python main.py --arg="--myflag" --arg="--myvar1=''" --arg="--myvar2 ''" --arg="--myvar3 a b c" --arg="--myvar4 'a b c'" --arg=--myvar5=zxcv
['main.py', '--arg=--myflag', "--arg=--myvar1=''", "--arg=--myvar2 ''", '--arg=--myvar3 a b c', "--arg=--myvar4 'a b c'", '--arg=--myvar5=zxcv']
Argument: --myflag
 -> '--myflag'
 => ['--myflag', None]
Argument: --myvar1=''
 -> '--myvar1'
 -> ''
 => ['--myvar1', '']
Argument: --myvar2 ''
 -> '--myvar2'
 -> ''
 => ['--myvar2', '']
Argument: --myvar3 a b c
 -> '--myvar3'
 -> 'a'
 -> 'b'
 -> 'c'
 => ['--myvar3', 'a', 'b', 'c']
Argument: --myvar4 'a b c'
 -> '--myvar4'
 -> 'a b c'
 => ['--myvar4', 'a b c']
Argument: --myvar5=zxcv
 -> '--myvar5'
 -> 'zxcv'
 => ['--myvar5', 'zxcv']
```

```python
import click
import shlex
import sys


@click.command()
@click.option('--arg', multiple=True, type=str)
def main(arg):
    print(sys.argv)
    for a in arg:
        print(f'Argument: {a}')
        tokens = []
        for token in shlex.shlex(a, punctuation_chars=True):
            kv = token.split('=', maxsplit=1)
            if len(kv) == 1:
                tokens.append(shlex.split(token)[0])
                print(' ->', repr(tokens[-1]))
            else:
                tokens.append(kv[0])
                tokens.append(shlex.split(kv[1])[0])
                print(' ->', repr(tokens[-2]))
                print(' ->', repr(tokens[-1]))
        if len(tokens) == 1:
            tokens.append(None)
        print(' =>', tokens)


if __name__ == '__main__':
    main()
```

Reply:
To handle the case of `myvar3`, which accepts multiple subsequent values for a single argument, we need to also extend the accepting type of arg values to be either None, a string, and *a list of strings* in the manager & agent.